cmd/wallet-web/src/router/index.js

```diff
@@ -6,15 +6,29 @@ SPDX-License-Identifier: Apache-2.0
 
 import { createRouter, createWebHistory } from 'vue-router';
 import store from '@/store';
+import { createKeyPair, gnapRequestAccess } from '@/mixins';
 import routes from './routes';
+import { computed } from 'vue/dist/vue.esm-bundler';
 
 const router = createRouter({
   history: createWebHistory(__webpack_public_path__),
   routes,
 });
 
-router.beforeEach((to, from, next) => {
+router.beforeEach(async (to, from, next) => {
   store.dispatch('agent/flushStore');
+  if (to.path === '/gnap') {
+    const gnapKeypair = await createKeyPair();
+    store.dispatch('updateGnapKeyPair', gnapKeypair);
+
+    const signer = { SignatureVal: gnapKeypair };
+    const gnapAuthServerURL = computed(() => store.getters['hubAuthURL']);
+
+    const resp = await gnapRequestAccess(signer, gnapAuthServerURL);
```

Comment:
So here we are going to save properties from `resp` to Vuex/local storage and call `GNAPClient.continue()`, right? Can we add a TODO with a corresponding issue in this case?